### ISSUE TITLE
go-fuzz-build using go installed via snap gives 'cannot find package "go-fuzz-dep"'

### ISSUE THREAD

**Issue by pwaller** [dvyukov/go-fuzz#200]
I'm on Ubuntu 18.04, and Go is installed using `snap install go`. `go-fuzz-build` gives the following error:

```
/tmp/go-fuzz-build083832919/gopath/src/github.com/pwaller/testfuzz/fuzz.go:1: cannot find package "go-fuzz-dep" in any of:
	/snap/go/3039/src/go-fuzz-dep (from $GOROOT)
	/tmp/go-fuzz-build083832919/gopath/src/go-fuzz-dep (from $GOPATH)
```

I was unable to make an easy reproducer using Docker because I guess snap can't easily be put in docker.

I can see that `/tmp/go-fuzz-build083832919/goroot/src/go-fuzz-dep` does exist, but from the error message it looks like it is not being considered for some reason.

**Comment by dvyukov** [comment]
I don't know what to say. I don't know what is snap, what it does and how/why it's different. Normal `go get` should work.

**Comment by dgryski** [comment]
This issue should be filed upstream with the Snap Developers.

**Comment by dgryski** [comment]
It's not clear where the appropriate "upstream" is for this issue.

Ping @mwhudson

**Comment by mwhudson** [comment]
Thanks for the ping, I'll have a look some time today hopefully.

**Comment by pwaller** [comment]
Some additional reproduction instructions, in case it isn't obvious. I made a package called `testfuzz` in my `GOPATH`, disabled modules with `GO111MODULE=off`, and defined a `func Fuzz(data []byte) int { return 1 }` function in a `testfuzz` package.

Then I tried building said package with `go-fuzz-build`, and hit the error I reported above.

**Comment by josharian** [comment]
Want to try again? Large parts of go-fuzz-build have been updated recently.

**Comment by pwaller** [comment]
Still getting this.

```
GO111MODULE=off GOPATH=$HOME/.local go-fuzz-build github.com/pwaller/testfuzz
failed to execute go build: exit status 1
fuzz.go:1: cannot find package "go-fuzz-dep" in any of:
	/snap/go/3540/src/go-fuzz-dep (from $GOROOT)
	/tmp/go-fuzz-build969880179/gopath/src/go-fuzz-dep (from $GOPATH)
```

```go
// ~/.local/src/github.com/pwaller/testfuzz$ cat fuzz.go 
package testfuzz

func Fuzz(data []byte) int {
	return 1
}
```

Tested with `go1.12.2` from snap, and go-fuzz 9cfa592d57929860bc951b39c3ca41a2a9d4c919.

**Comment by josharian** [comment]
Hmm. What does `go list std` say? (Does it spew ~194 packages, or give an error?) What about `go list github.com/dvyukov/go-fuzz/go-fuzz-dep`?

**Comment by mwhudson** [comment]
Ah the problem is that the wrapper around the go executable in the snap sets GOROOT:

```
mwhudson@ringil:/opt/opensource/gopath/src/testfuzz$ cat /snap/go/current/gowrapper 
#!/bin/bash
export GOROOT="$SNAP"
exec $SNAP/bin/go "$@"
```

I should change that to only set GOROOT if not already set in the environment.

**Comment by mwhudson** [comment]
I've just released a fix for this to the stable, 1.12/stable and 1.11/stable channels. The test case from @pwaller works for me now, please let me know if it does or does not for you.

**Comment by pwaller** [comment]
It works now, thanks all.
